src/unix/apple/disk.rs

```diff
@@ -400,9 +400,19 @@ unsafe fn new_disk(
 
     let available_space = get_available_volume_space(disk_props);
 
-    let file_system = IntoIterator::into_iter(c_disk.f_fstypename)
-        .filter_map(|b| if b != 0 { Some(b as u8) } else { None })
-        .collect();
+    let file_system = {
```

Comment:
Isn't it doing exactly the same but with more code and allocations?

Reply:
Exactly the same amount of allocations. `OsString::from_vec` takes ownership of the vec.

I'd argue that the previous code was not correct. The previous code would convert a slice `['a', 'b', '\0', 'Z']` into `"abZ"` (doesn't stop at the first zero code point).

Reply:
Oh indeed. Good catch!